src/checkers/inference/InferenceLauncher.java

```diff
@@ -152,12 +152,13 @@ public void typecheck(String [] javaFiles) {
     public void infer() {
         printStep("Inferring", outStream);
         final String java = PluginUtil.getJavaCommand(System.getProperty("java.home"), outStream);
+        final boolean isJava8 = PluginUtil.getJreVersion() == 8;
```

Comment:
It seems a bit odd to check for Java 8 both here and in the `inference-dev` script.
Couldn't you just check whether the values for those properties are non-empty?

I was also concerned that https://github.com/typetools/checker-framework-inference/blob/master/src/checkers/inference/InferenceDevelLauncher.java#L84 builds an invalid argument when the `annotated.jdk` property is empty. But then I realized that that method isn't called here if you're on Java 8.

So how about changing `getInferenceCompilationBootclassPath` to just return the path to use and here you check whether that is non-empty, and, if so, you add the `-Xbootclasspath/p:` part here.

Reply:
It seems https://github.com/typetools/checker-framework-inference/blob/master/src/checkers/inference/InferenceLauncher.java#L406 is not using an environmental variable. I'm considering adding a `jdkPaths` to the `inference` script, but then we might need to do the JDK version check from `PluginUtil.getJdkJarName()` in the script instead.

Reply:
In `checker-framework` can you look at the other uses of `PluginUtil.getJdkJarName()`, in particular in CheckerMain, and suggest a nice solution that works for both Java 8 and 9+, in a nice way?
If you look at CheckerMain, you similarly see that the `jdkJar` variable isn't used on 9+. So making this more uniform and well documented would be good.
I agree that introducing another environment variable wouldn't be very nice.

Reply:
I have tried making some changes to `PluginUtil.getJdkJarName()` (https://github.com/typetools/checker-framework/compare/master...zcai1:zcai/change-getJdkJarPrefix?expand=1); however, I think it also breaks the file loading pattern in `CheckerMain`.

This would be the change I suggest:
https://github.com/typetools/checker-framework-inference/pull/151/commits/b3f7a759837e9ea7c43052a7ad92aafc9c133ecc

`PluginUtil.getJdkJarName()` returns `jdk11.jar` under JRE11, so the function should simply return an empty string if the file is missing.